src/Entities/AbstractEntity.php

```diff
@@ -0,0 +1,182 @@
+                $getValue = static function ($value) use ($type) {
+                    $data = $value;
+
+                    if ($value instanceof JsonSerializable) {
+                        $data = $value->jsonSerialize();
+                    }
+
+                    return new $type($data);
+                };
+            } else {
+                $errors[] = sprintf('Missing class: %s', $type);
+
+                continue;
+            }
+
+            if ($attribute->isArray) {
+                $property->setValue($this, array_map(static function ($value) use ($getValue) {
+                    return $getValue($value);
+                }, $data[$name]));
+            } else {
+                $property->setValue($this, $getValue($data[$name]));
+            }
+        }
+
+        if (! empty($errors)) {
+            throw new EntitiesException(static::class . ': ' . implode(', ', $errors));
+        }
+    }
+
+    public function findPropertyAttribute(ReflectionProperty $property): ?Property
+    {
+        $attributes = $property->getAttributes();
+
+        foreach ($attributes as $attribute) {
+
+            $newInstance = $attribute->newInstance();
+            if ($newInstance instanceof Property) {
+                return $newInstance;
+            }
+        }
+
+        return null;
+    }
+
+    public function jsonSerialize(): array
+    {
+        $fields = [];
+
+        $reflectionClass = new ReflectionClass(static::class);
+        $properties = $reflectionClass->getProperties();
+        foreach ($properties as $property) {
+            $attribute = $this->findPropertyAttribute($property);
+
+            if (! $attribute instanceof Property) {
+                continue;
+            }
+            $name = $attribute->name ?? $property->getName();
+
+            $property->setAccessible(true);
+
+            if ($attribute->required !== false || $property->getValue($this) !== null) {
+                $fields[$name] = $property->getValue($this);
+                if ($fields[$name] instanceof JsonSerializable) {
+                    $fields[$name] = $fields[$name]->jsonSerialize();
+                } elseif ($fields[$name] instanceof DateTimeInterface) {
+                    $fields[$name] = $fields[$name]->format(DateTimeInterface::ATOM);
+                } elseif (is_array($fields[$name])) {
+                    $fields[$name] = array_map(static function ($value) {
+                        if ($value instanceof JsonSerializable) {
+                            return $value->jsonSerialize();
+                        }
+
+                        return $value;
+                    }, $fields[$name]);
+                }
+            }
+        }
+
+        return $fields;
+    }
+
+    public function __call(string $method, array $params): mixed
+    {
+        $property = lcfirst(substr($method, 3));
+        $prefix = substr($method, 0, 3);
+
+        if (! in_array($prefix, ['get', 'set'])) {
```

Comment:
Now that the PHP minimum version is 8 can't you use match ?

Reply:
Anyway, this in_array makes the final return useless.

Reply:
i think  if (! in_array($prefix, ['get', 'set'])) is useless you right thx !!